### ISSUE TITLE
Print all assigned shows other people signature

### ISSUE THREAD

**Issue by raphbeaulieu** [grokability/snipe-it#14734]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

Hi,

Every time I print the list of products assigned to a user, the report displays the signature of the last user who signed this material.

The \resources\views\users\print.blade.php file look like this (i've never modified the file) :
`                            @if (($accessory->assetlog->first()) && ($accessory->assetlog->first()->accept_signature!=''))
                            <img style="width:auto;height:100px;" src="{{ asset('/') }}display-sig/{{ $accessory->assetlog->first()->accept_signature }}">
                            @endif
`

I ran this commands without success :
-   composer dump-autoload
-   php artisan cache:clear 
-   php artisan config:clear
-   php artisan route:clear
-   php artisan debugbar:clear
-   php artisan event:clear
-   php artisan view:clear
-   php artisan optimize:clear
-   php artisan clear-compiled

### Reproduction steps

1. Go to user info page
2. Press the Print all assigned button

### Expected behavior

I expect to see the signatures of the selected user

### Screenshots

It shows the signatures of other users rather than the selected user

![image](https://github.com/snipe/snipe-it/assets/106820939/d8168e32-88c9-4bf4-ad74-49c71dbaf959)

### Snipe-IT Version

v6.4.1 - build 13458 (master)

### Operating System

Windows Server 2019 Standard

### Web Server

IIS 10.0.17763.1

### PHP Version

8.2.3

### Operating System

Windows 11 Business

### Browser

Google Chrome

### Version

V124

**Comment by marcusmoore** [comment]
Thanks for reporting this @raphbeaulieu. I'll check it out.

**Comment by raphbeaulieu** [comment]
I'm still having the same issue after i upgraded to version 7.0.3.

![image](https://github.com/snipe/snipe-it/assets/106820939/9aeef75e-570b-4f8a-a6f1-621b574911fb)

**Comment by marcusmoore** [comment]
Yeah...we don't have a fix for this merged yet but it's still on our radar.

**Comment by lukaspetrik** [comment]
If anyone is facing this issue, I have created a **quick workaround**. This workaround displays the signature from the most recent accessory action related to the current user, rather than most recent action related to all users of that accessory. The $accessory->assetlog contains accessory actions of all users, and using first() from that list always results in showing the signature of the most recent user instead of the current one. Replacing code at line ~300 with the following fixes it:
```
<td>{{ $accessory->pivot->created_at }}</td>
<td>
    @foreach ($accessory->assetlog as $logaction)
        @if (($logaction->targetType() == 'user') && ($show_user['id'] == $logaction->target->id) && ($logaction->accept_signature!=''))
            <img style="width:auto;height:100px;" src="{{ asset('/') }}display-sig/{{ $logaction->accept_signature }}">
            @break
        @endif
    @endforeach
</td>
```
Similar issue is there for consumables etc., but using this workaround can fix signatures there as well.

**Comment by snipe** [comment]
@marcusmoore can you take a look and see if this is something we can patch to develop?
